ale_linters/cpp/clangtidy.vim

```diff
@@ -35,17 +43,45 @@ endfunction
 function! ale_linters#cpp#clangtidy#GetCommand(buffer) abort
     let l:checks = join(ale#Var(a:buffer, 'cpp_clangtidy_checks'), ',')
     let l:build_dir = s:GetBuildDirectory(a:buffer)
-
     " Get the extra options if we couldn't find a build directory.
     let l:options = empty(l:build_dir)
-    \   ? ale#Var(a:buffer, 'cpp_clangtidy_options')
-    \   : ''
-
-    return ale#Escape(ale_linters#cpp#clangtidy#GetExecutable(a:buffer))
-    \   . (!empty(l:checks) ? ' -checks=' . ale#Escape(l:checks) : '')
-    \   . ' %s'
-    \   . (!empty(l:build_dir) ? ' -p ' . ale#Escape(l:build_dir) : '')
-    \   . (!empty(l:options) ? ' -- ' . l:options : '')
+                \   ? ale#Var(a:buffer, 'cpp_clangtidy_options')
+                \   : ''
+
+    if !ale_linters#cpp#clangtidy#IsHeader(a:buffer)
```

Comment:
I'm really not a fan of massive `if ... else` blocks like this. It's better if you can just progressively build options and return a command at the end, so the code just flows top to bottom.